posts/2021-08-06-21009-beta.adoc

```diff
@@ -0,0 +1,179 @@
+A new method was added to the `ServerEndpointControlMBean`:
+
+`public boolean isActive(String targets);`
+
+The `targets` parameter is a comma separated list of endpoint names. 
+
+This mirrors the existing `isPaused` method. It will return true only if all of the endpoints listed exist, are started, and are not paused.
+
+[source, java]
+----
+// Check if the defaultHttpEndpoint is active
+boolean isEndpointActive = mbean.isActive("defaultHttpEndpoint");
+----
+
+When an invalid endpoint is passed to `isActive`, the method returns `False`, even if the other endpoints are all valid.
+
+[source, java]
+----
+assertFalse("False should be returned when an invalid endpoint is passed to isActive", mbean.isActive("inValidTargetName!"));
+
+// create a list of valid endpoints
+String targets = createHttpEndpoints(4);
+// add an invalid endpoint
+targets += ",invalidEndpoint";
+
+// Check if the defaultHttpEndpoint is active
+// isEndpointActive should be False
+boolean isEndpointActive = mbean.isActive(targets);
+----
+
+When a valid endpoint is passed to `isActive`, the method returns `False` if any of the endpoints are paused.
+
+[source, java]
+----
+// pause active endpoint
+try {
+    mbean.pause("defaultHttpEndpoint");
+} catch (MBeanException e) {
+    fail("Got unexpected exception from attempting to pause a endpoint with a valid target 'defaultHttpEndpoint':" + e.getCause());
+}
+
+// Check if the defaultHttpEndpoint is active
+// isEndpointActive should be False as endpoint is paused
+boolean isEndpointActive = mbean.isActive(targets);
+
+// Resume all targets
+mbean.resume();
+----
+
+For more information about `ServerEndpointControlMBean` methods, refer to the link:https://www.ibm.com/docs/en/was-liberty/nd?topic=SSAW57_liberty/com.ibm.websphere.javadoc.liberty.doc/com.ibm.websphere.appserver.api.kernel.service_1.0-javadoc/com/ibm/websphere/kernel/server/ServerEndpointControlMBean.html[API documentation].
+
+
+=== Try it now 
+
+To try out these features, just update your build tools to pull the Open Liberty All Beta Features package instead of the main release. The beta works with Java SE 15, Java SE 11, or Java SE 8.
+
+If you're using link:{url-prefix}/guides/maven-intro.html[Maven], here are the coordinates:
+
+[source,xml]
+----
+<dependency>
+  <groupId>io.openliberty.beta</groupId>
+  <artifactId>openliberty-runtime</artifactId>
+  <version>21.0.0.9-beta</version>
+  <type>pom</type>
+</dependency>
+----
+
+Or for link:{url-prefix}/guides/gradle-intro.html[Gradle]:
+
+[source,gradle]
+----
+dependencies {
+    libertyRuntime group: 'io.openliberty.beta', name: 'openliberty-runtime', version: '[21.0.0.9-beta,)'
+}
+----
+
+Or take a look at our link:{url-prefix}/downloads/#runtime_betas[Downloads page].
+
```

Comment:
[#mp4.1]
== MicroProfile 4.1 Beta Features
MicroProfile 4.1 was originally released in the Open Liberty Beta 21.0.0.8-beta and it is also in this beta. link:https://openliberty.io/blog/2021/07/12/logging-format-21008-beta.html#mp4_1[Here] has more information regarding MicroProfile 4.1.

Reply:
Added!